test/task.cc

```diff
@@ -204,6 +257,45 @@ TEST(Task, FailTerminated) {
       // if we capture it by copy we can't call 'get()' because that
       // is a non-const function.
       [future = &future]() { future->get(); },
+      ThrowsMessage<std::runtime_error>(StrEq("error")));
+}
+
+TEST(Task, FailTerminatedCatch) {
+  MockFunction<void()> fail;
+
+  EXPECT_CALL(fail, Call())
+      .Times(1);
+
+  auto e = [&]() -> Task::Of<int>::Raises<
+                     std::runtime_error>::Catches<std::runtime_error> {
```

Comment:
Doesn't seem like we are exercising `Catches` here? Is this test redundant with any others once we drop `Catches<std::runtime_error>`?